### PR TITLE
refactor(test): replace Dependencies annotation with IntegrationEnvironment

### PR DESCRIPTION
**Pull request by Sabo2k**
This is my first contribution to an Open Source Project 🙂

I replaced the annotation ```Dependencies``` in ```BlockTest```, ```DamageEventTest```, ```RestorationTest``` and ```RegenTest``` with ```IntegrationEnvironment```. 

For more information, see <a href="https://github.com/MovingBlocks/Terasology/issues/5087">#5087</a>

### CODE REVIEW

File: src/test/java/org/terasology/module/health/BlockTest.java [hunk 1]
```diff
@@ -28,7 +28,7 @@
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
 @ExtendWith(MTEExtension.class)
-@Dependencies("Health")
+@IntegrationEnvironment(dependencies = { "Health" })
 @Tag("MteTest")
```
Comment (jdrueckert) [inline src/test/java/org/terasology/module/health/BlockTest.java]: As mentioned in https://github.com/MovingBlocks/Terasology/issues/5087
> the class annotation `@IntegrationEnvironment` [...] will apply `MTEExtension`, mark it with an `MteTest` JUnit tag

Accordingly, you can remove the `@ExtendWith` and `@Tag` annotations as well.

Same for all other changes

**Review by jdrueckert — CHANGES_REQUESTED** [review]

**Comment by skaldarnar** [comment]
@Sabo2k please have a look at the reported CheckStyle warnings in this PR (unused imports and the code style when using annotations). Once those are resolved, we can merge this 👍

**Comment by Sabo2k** [comment]
When I cloned my forked Health-repo, I couldn't run all tests and static code checks for this module using 
```
gradlew :module:Health:check
``` 

because it didn't contain a ```build.gradle``` file. 

Do you guys think it would be a good idea to add one to the repository? 🙂

**Comment by jdrueckert** [comment]
> because it didn't contain a `build.gradle` file.

This is expected based on how our CI jobs are set up.
You can run checks locally for specific modules from your workspace root using `./gradlew :modules:<module>:check`, e.g. `./gradlew :modules:Health:check`

**Review by jdrueckert — APPROVED** [review]
Looks good to me :+1:

**Pull request merged**